common/persistence/class.SqlKvDriver.php

```diff
@@ -181,25 +182,33 @@ public function del($id)
     /**
      * Increment existing value
      * @param string $id
-     * @return mixed
+     * @return boolean
+     *
+     * @throws common_Exception
      */
     public function incr($id)
     {
-        $params = [':id' => $id];
-        $statement = 'UPDATE kv_store SET kv_value = kv_value + 1 WHERE kv_id = :id';
-        return $this->sqlPersistence->exec($statement, $params);
+        $value = filter_var($this->get($id), FILTER_VALIDATE_INT);
+        if ($value !== false) {
+            return $this->set($id, $value + 1);
+        }
+        return false;
     }
 
     /**
      * Decrement existing value
      * @param $id
-     * @return mixed
+     * @return boolean
+     *
+     * @throws common_Exception
      */
     public function decr($id)
     {
-        $params = [':id' => $id];
-        $statement = 'UPDATE kv_store SET kv_value = kv_value - 1 WHERE kv_id = :id';
-        return $this->sqlPersistence->exec($statement, $params);
+        $value = filter_var($this->get($id), FILTER_VALIDATE_INT);
```

Comment:
Example:
try to run 1000 test takers at the same time I'm pretty sure that it will lead us to the situation when one updater will overwrite another value, I mean that when 1 php process gets value - another parallel process gets the same value, so on update these processes will write the same incorrect value.
1.0 get 10 delivery executions
1.1 get 10 de
2.0 10-1 write 9 to db
2.1 write 9 to db as well, but there should be 8 already

Reply:
agreed, will revert to previous implementation and add spanner flavor